airflow/api_connexion/openapi/v1.yaml

```diff
@@ -2112,12 +2112,13 @@ paths:
 
   /roles:
     get:
+      deprecated: true
       summary: List roles
       description: |
         Get a list of roles.
 
         *New in version 2.1.0*
-      x-openapi-router-controller: airflow.api_connexion.endpoints.role_and_permission_endpoint
+      x-openapi-router-controller: airflow.auth.managers.fab.api_endpoints.role_and_permission_endpoint
```

Comment:
I think we need to add a layer in between. We dont want these endpoints to be available if the auth manager used is not FAB auth manager

Reply:
Said another way, we want to make sure these fail gracefully in that scenario, right?

Reply:
Correct :) If the auth manager used is not FAB auth manager, 404 would be ideal

Reply:
ah yes you're right.